src/libsyntax/parse/diagnostics.rs

```diff
@@ -1074,16 +1074,17 @@ impl<'a> Parser<'a> {
         Err(err)
     }
 
-    crate fn eat_incorrect_doc_comment(&mut self, applied_to: &str) {
+    crate fn eat_incorrect_doc_comment_for_arg_type(&mut self) {
         if let token::DocComment(_) = self.token.kind {
             let mut err = self.diagnostic().struct_span_err(
                 self.span,
-                &format!("documentation comments cannot be applied to {}", applied_to),
+                "documentation comments cannot be applied to a method argument's type",
             );
             err.span_label(self.span, "doc comments are not allowed here");
             err.emit();
             self.bump();
-        } else if self.token == token::Pound && self.look_ahead(1, |t| {
+        }
+        else if self.token == token::Pound && self.look_ahead(1, |t| {
```

Comment:
```suggestion
        } else if self.token == token::Pound && self.look_ahead(1, |t| {
```
Nit: unidiomatic `else` formatting here and in `fn check_fn_decl`.